meta.go

```diff
@@ -147,6 +149,34 @@ type Device struct {
 	CountMax int64 `bson:"countmax"`
 }
 
+// DeploymentType defines a deployment type.
+type DeploymentType string
+
+const (
+	DeploymentStateless DeploymentType = "stateless"
+	DeploymentStateful  DeploymentType = "stateful"
+)
+
+// ServiceType defines a service type.
+type ServiceType string
+
+const (
+	ServiceCluster      ServiceType = "cluster"
+	ServiceLoadBalancer ServiceType = "loadbalancer"
+	ServiceExternal     ServiceType = "external"
```

Comment:
just wondering how we are going to use `ServiceLoadBalancer` and `ServiceExternal`

Reply:
We just create the k8s service with what the charm specifies.
It's up to the user to have their cluster correctly configured. We can figure out if there's any way to do some pre-checks of the cluster at deploy time, but that doesn't affect the metadata definition.